lib/node_modules/@stdlib/array/complex64/test/test.for_each.js

```diff
@@ -0,0 +1,175 @@
+var isFunction = require( '@stdlib/assert/is-function' );
+var reinterpret64 = require( '@stdlib/strided/base/reinterpret-complex64' );
+var isComplex64 = require('@stdlib/assert/is-complex64');
+var Complex64Array = require( './../lib' );
+
+
+// TESTS //
+
+tape( 'main export is a function', function test( t ) {
+	t.ok( true, __filename );
+	t.strictEqual( typeof Complex64Array, 'function', 'main export is a function' );
+	t.end();
+});
+
+tape( 'attached to the prototype of the main export is a `forEach` method', function test( t ) {
+	t.strictEqual( hasOwnProp( Complex64Array.prototype, 'forEach' ), true, 'has property' );
+	t.strictEqual( isFunction( Complex64Array.prototype.forEach ), true, 'has method' );
+	t.end();
+});
+
+tape( 'the method throws an error if invoked with a `this` context which is not a complex number array instance', function test( t ) {
+	var values;
+	var arr;
+	var i;
+
+	arr = new Complex64Array( 5 );
+
+	values = [
+		'5',
+		5,
+		NaN,
+		true,
+		false,
+		null,
+		void 0,
+		{},
+		[],
+		function noop() {}
+	];
+	for ( i = 0; i < values.length; i++ ) {
+		t.throws( badValue( values[i] ), TypeError, 'throws an error when provided '+values[i] );
+	}
+	t.end();
+
+	function badValue( value ) {
+		return function badValue() {
+			return arr.forEach.call( value, fcn );
+		};
+	}
+
+	function fcn( v ) {
+		if ( !isComplex64( v ) ) {
+			t.throws( 'should be a complex number ' );
+		}
+	}
+});
+
+tape( 'the method throws an error if provided a first argument which is not a function', function test( t ) {
+	var values;
+	var arr;
+	var i;
+
+	arr = new Complex64Array( 10 );
+
+	values = [
+		'5',
+		3.14,
+		NaN,
+		true,
+		false,
+		null,
+		void 0,
+		{},
+		[]
+	];
+	for ( i = 0; i < values.length; i++ ) {
+		t.throws( badValue( values[i] ), TypeError, 'throws an error when provided '+values[i] );
+	}
+	t.end();
+
+	function badValue( value ) {
+		return function badValue() {
+			return arr.forEach( value );
+		};
+	}
+});
+
+tape( 'the method will not invoke callback function if provided an empty complex number array', function test( t ) {
+	var arr;
+
+	arr = new Complex64Array();
+	arr.forEach( fcn );
+
+	t.end();
+
+	function fcn() {
+		t.fail( 'should not be invoked' );
+	}
+});
+
+tape( 'the method returns `undefined`', function test( t ) {
+	var arr;
+	var out;
+
+	arr = new Complex64Array( [ 1.0, 2.0, 3.0, 4.0 ] );
+	out = arr.find( fcn );
+
+	t.strictEqual( out, void 0, 'returns expected value' );
+	t.end();
+
+	function fcn( v ) {
+		if ( !isComplex64( v ) ) {
+			t.throws( 'should be a complex number' );
+		}
+	}
+});
+
+tape( 'the method invokes the function for each element in an array', function test( t ) {
+	var arr;
+	var out;
+
+	out = [];
+	arr = new Complex64Array( [ 1.0, 1.0, 2.0, -2.0 ] );
+	arr.forEach( fcn );
+
+	t.deepEqual( reinterpret64( out, 0 ), reinterpret64( out, 0 ), 'returns expected value' );
+	t.end();
+
+	function fcn( v ) {
+		out.push( v );
+	}
```

Comment:
@Jaysukh-409  I'm not sure what you are trying to test here. I am surprised this test even runs. `reinterpret64` is not intended to operate on generic arrays, so provided `out` does not seem correct.

Reply:
And also, `reinterpret64( out, 0 ), reinterpret64( out, 0 )` --> I'd expect these two things to be equal. You are testing that `out` equals `out`.